### DISCUSSION TITLE
Possible bug in trailer recognition?

### DISCUSSION THREAD

**Discussion by zehnerGIT** [jellyfin/jellyfin#7566] [Q&A] [answered]
https://github.com/jellyfin/jellyfin/blob/d5cc2ad6a8cdaa9c3d3d8c288643a18eaa2dc6f1/MediaBrowser.Providers/Plugins/Tmdb/TmdbUtils.cs#L84

I don't understand the original code here.

```
public static bool IsTrailerType(Video video)
        {
            return video.Site.Equals("youtube", StringComparison.OrdinalIgnoreCase)
                   && (!video.Type.Equals("trailer", StringComparison.OrdinalIgnoreCase)
                       || !video.Type.Equals("teaser", StringComparison.OrdinalIgnoreCase));
        }
```

It seems, that someone wants to check if the video is a trailer or teaser, but the code does the complete opposite, it adds almost everything (e.g. video.Type == "trailer", video.Type == "featurette", ...). If the intention was just to add real trailers, then something like

```
public static bool IsTrailerType(Video video)
{
    return "youtube".Equals(video.Site, StringComparison.OrdinalIgnoreCase)
           && ("trailer".Equals(video.Type, StringComparison.OrdinalIgnoreCase)
               || "teaser".Equals(video.Type, StringComparison.OrdinalIgnoreCase));
}
```
should work.
I've changed the order of the .Equals to prevent NullReferenceExceptions, because https://developers.themoviedb.org/3/movies/get-movie-videos shows Site and Type as optional, but that should not be necessary, because they seem to always have values (otherwise you would have many user complaints about exceptions). If you already have complaints, then you can add a `string.IsNullOrWhitespace(video.Type) ||` to the bracket to handle NULLs always as trailer as default

**Comment by crobibero** [comment, marked as answer]
This has already been fixed in https://github.com/jellyfin/jellyfin/pull/7551

But it has not been pushed to master yet.
